### PR TITLE
feat: support ReactiveUI 24 and updated analyzers

### LINKED ISSUE #469: [Feature]: Add support for both ReactiveUI 24 API profiles
### Is your feature request related to a problem?

ReactiveUI.SourceGenerators currently documents compatibility with ReactiveUI 23.2.28+, but ReactiveUI 24 introduces two distinct API profiles:

- The default Primitives-based profile, using the `ReactiveUI` package family.
- The System.Reactive-compatible profile, using the `ReactiveUI.Reactive` package family.

Projects upgrading to ReactiveUI 24 need to be able to continue using ReactiveUI.SourceGenerators regardless of which ReactiveUI 24 API profile they select.

### Describe the solution you'd like

Please add official ReactiveUI 24 compatibility for both supported API profiles:

1. The default Primitives-based ReactiveUI 24 profile.
2. The System.Reactive-compatible ReactiveUI 24 `.Reactive` profile.

All currently supported source generators and attributes should compile and operate correctly when the consuming project uses either profile.

### Describe alternatives you've considered

The current alternatives are to remain on ReactiveUI 23 or replace source-generated members with manually written implementations.

Neither alternative is suitable for projects adopting ReactiveUI 24.

### Suggestions on how to achieve the feature

Keep a single `ReactiveUI.SourceGenerators` package and detect the available ReactiveUI 24 API profile from the consuming compilation.

Centralize profile-specific namespaces and type mappings, then generate code for either:

- The default Primitives-based `ReactiveUI` profile.
- The System.Reactive-compatible `ReactiveUI.Reactive` profile.

Add compile-time integration tests for both package families to prevent compatibility regressions.

### Additional context

Support should not be limited to only one of the ReactiveUI 24 package families. Both the default Primitives-based profile and the System.Reactive-compatible `.Reactive` profile should be treated as supported configurations.

### PR DESCRIPTION
**Pull request by ChrisPulman**
## What kind of change does this PR introduce?

Feature and maintenance update that adds ReactiveUI 24 package-profile compatibility and remediates diagnostics from the StyleSharp, PerformanceSharp, and SecuritySharp 3.40 analyzer update.

Closes #469 

## What is the new behavior?

The generators detect ReactiveUI 23 legacy, ReactiveUI 24 base/Primitives, and ReactiveUI 24 Reactive profiles, emit the matching APIs, and use lifecycle-safe serialized disposable composition for generated WinForms hosts. The full solution is analyzer-clean.

## What is the current behavior?

The previous implementation targets the legacy ReactiveUI surface, relies on System.Reactive-specific command output, and reports analyzer errors and warnings after the analyzer package update. Host composition also has disposal and terminal-state edge cases.

## Checklist
- [x] Tests have been added or updated (for bug fixes / features) 
- [x] Docs have been added or updated (for bug fixes / features)
- [x] Changes target the main branch
- [x] PR title follows [Conventional Commits](https://www.conventionalcommits.org/) 

## Additional information

- Release solution build: 0 warnings, 0 errors.
- TUnit / Microsoft.Testing.Platform: 302 tests passed on each of net8.0, net9.0, and net10.0 (906 framework-specific executions).
- Coverage: 63.32% line, 62.16% branch.
- Analyzer diagnostics were remediated in code; no suppression changes were used to resolve build findings.

### CODE REVIEW

File: src/Directory.Packages.props [hunk 1]
```diff
@@ -25,12 +28,14 @@
...
     <PackageVersion Include="Avalonia.Desktop" Version="12.1.0" />
     <PackageVersion Include="Microsoft.Extensions.Logging.Debug" Version="10.0.10" />
+    <PackageVersion Include="DynamicData" Version="9.4.33" />
     <PackageVersion Include="Microsoft.NET.Test.Sdk" Version="18.7.0" />
     <PackageVersion Include="System.Formats.Asn1" Version="10.0.10" />
     <PackageVersion Include="PublicApiGenerator" Version="11.5.4" />
     <PackageVersion Include="Splat" Version="20.2.0" />
-    <PackageVersion Include="ReactiveUI" Version="23.2.28" />
-    <PackageVersion Include="ReactiveUI.Maui" Version="23.2.28" />
+    <PackageVersion Include="ReactiveUI" Version="24.0.0" />
```
Comment (glennawatson) [inline src/Directory.Packages.props:36]: This is the conflict with `main` - the only one in the PR. Needs a rebase/merge, then re-verify the version set here still matches what `main` picked up from the dependency bumps.

**Review by glennawatson — COMMENTED** [review]
Can't merge as-is - it conflicts with `main`. The code itself is in decent shape; the problem is mostly size.

### Blocking

- **Conflicts with `main`.** Exactly one file: `src/Directory.Packages.props`. Trivial to resolve, but it needs doing before anything else.

### Scope

217 files, +12,809 / -7,168. That is two unrelated changes in one PR:

1. ReactiveUI 24 profile support (the actual feature - `ReactiveUiApi`, `ReactiveUiIntegration`, `ContextExtensions.GetReactiveUiIntegration`, and the generator/template updates that consume them).
2. A repo-wide analyzer 3.40 remediation - StyleCop removal, `.editorconfig` rewrite (+2,689 / -518), and mechanical edits across every test, sample app and helper.

Splitting those makes both reviewable. Right now the feature is roughly 15 files buried in 200 of churn, and a reviewer can't tell the two apart.

### What looks good

- The profile split is clean. `ReactiveUiIntegration` is a small equatable `readonly record struct`, so hanging it off `CompilationProvider` doesn't defeat incremental caching downstream.
- Detection order is right: `ReactiveUI.Reactive.ReactiveCommand` first, then legacy `ReactiveUI.ReactiveCommand` + `RxVoid` probe, then the referenced-assembly fallback.
- All three profiles are actually tested - `V24BaseUsesPrimitives...`, `V24ReactiveUsesReactiveNamespace...` and `V23UsesLegacyNamespace...`, the last against a synthesized `AssemblyVersion("23.2.28.0")` reference. Good; that path is easy to leave untested.

### Worth a second look

- The new `.gitattributes` (`*.cs text eol=lf`) is a repo-wide normalisation riding along in a feature PR. It will renormalise line endings on next checkout for anyone on Windows. Fine to do - just should be its own commit, and called out.
- `ReactiveUI` moves 23.2.28 -> 24.0.0 for the sample and test apps, so nothing builds against a real v23 package any more. The Legacy profile is covered only by the synthesized reference. Probably acceptable, but it does mean a v23 regression would not surface here.
- Coverage lands at 63.32% line / 62.16% branch. Low for a generator repo where the generated-output paths are the whole product.

**Review by glennawatson — APPROVED** [review]
